cli/cli.go

```diff
@@ -222,249 +242,293 @@ func list(c *cli.Context, t *toxiproxy.Client) {
+		return nil
 	}
 
 	if upstream {
-		addToxic("upstream")
+		err := addToxic("upstream")
+		if err != nil {
+			return err
+		}
 	}
 	// Default to downstream.
 	if downstream || (!downstream && !upstream) {
-		addToxic("downstream")
+		return addToxic("downstream")
 	}
+	return nil
 }
 
-func updateToxic(c *cli.Context, t *toxiproxy.Client) {
+func updateToxic(c *cli.Context, t *toxiproxy.Client) error {
 	proxyName := c.Args().First()
 	if proxyName == "" {
 		cli.ShowSubcommandHelp(c)
-		fatalf("Proxy name is required as the first argument.\n")
+		return errorf("Proxy name is required as the first argument.\n")
+	}
+	toxicName, err := getArgOrFail(c, "toxicName")
+	if err != nil {
+		return err
 	}
-	toxicName := getArgOrFail(c, "toxicName")
-	toxicAttributes := getArgOrFail(c, "attributes")
 
-	attributes := parseAttributes(toxicAttributes)
+	attributes := parseAttributes(c, "attributes")
 
 	p, err := t.Proxy(proxyName)
 	if err != nil {
-		fatalf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+		return errorf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
 	}
 
-	toxicity := parseToxicity(c, -1)
+	toxicity, err := parseToxicity(c, -1)
+	if err != nil {
+		return nil
+	}
 
 	_, err = p.UpdateToxic(toxicName, toxicity, attributes)
 	if err != nil {
-		fatalf("Failed to update toxic: %s\n", err.Error())
+		return errorf("Failed to update toxic: %s\n", err.Error())
 	}
 
 	fmt.Printf("Updated toxic '%s' on proxy '%s'\n", toxicName, proxyName)
+	return nil
 }
 
-func removeToxic(c *cli.Context, t *toxiproxy.Client) {
+func removeToxic(c *cli.Context, t *toxiproxy.Client) error {
 	proxyName := c.Args().First()
 	if proxyName == "" {
 		cli.ShowSubcommandHelp(c)
-		fatalf("Proxy name is required as the first argument.\n")
+		return errorf("Proxy name is required as the first argument.\n")
+	}
+	toxicName, err := getArgOrFail(c, "toxicName")
+	if err != nil {
+		return err
 	}
-	toxicName := getArgOrFail(c, "toxicName")
 
 	p, err := t.Proxy(proxyName)
 	if err != nil {
-		fatalf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
+		return errorf("Failed to retrieve proxy %s: %s\n", proxyName, err.Error())
 	}
 
 	err = p.RemoveToxic(toxicName)
 	if err != nil {
-		fatalf("Failed to remove toxic: %s\n", err.Error())
+		return errorf("Failed to remove toxic: %s\n", err.Error())
 	}
 
 	fmt.Printf("Removed toxic '%s' on proxy '%s'\n", toxicName, proxyName)
+	return nil
 }
 
-func parseAttributes(raw string) toxiproxy.Attributes {
+func parseAttributes(c *cli.Context, name string) toxiproxy.Attributes {
 	parsed := map[string]interface{}{}
-	keyValues := strings.Split(raw, ",")
-	for _, keyValue := range keyValues {
-		kv := strings.SplitN(keyValue, "=", 2)
-		if len(kv) != 2 {
-			fatalf("Attributes should be in format key1=value1,key2=value2\n")
+	args := c.StringSlice(name)
+
+	for _, raw := range args {
+		kv := strings.SplitN(raw, "=", 2)
+		if len(kv) < 2 {
+			continue
```

Comment:
I guess it's just a question of if something like `-a foo` should error or just be ignored. Right now it's just ignored, which seems fine to me, it will effectively just set a blank / 0 value in the toxic this way.